plugins/UM3NetworkPrinting/src/util.py

```diff
@@ -0,0 +1,37 @@
+
+from typing import Any, Dict, Optional
+
+from UM.Logger import Logger
+
+
+# Finds all machines (GlobalStacks) with the given network_key or group_name and update the metadata of those
+# machines with the key/values in the given metadata dict.
+# Note that the old authentication data "network_authentication_id" and "network_authentication_key" will be
+# deleted if found.
+def updateMetadataForNetworkGroupMachines(network_key: Optional[str], group_name: Optional[str],
```

Comment:
I think it makes more sense to put this in the DiscoverUM3Action.py file because it's a function about updating the metadata of printers with data discovered through zeroconf.